src/re_bytes.rs

```diff
@@ -1070,7 +1070,7 @@ pub trait Replacer {
     /// be beneficial to avoid finding sub-captures.
     ///
     /// In general, this is called once for every call to `replacen`.
-    fn no_expansion<'r>(&'r mut self) -> Option<Cow<'r, [u8]>> {
```

Comment:
Hmmm I'd actually rather have these lifetimes. I don't agree with all flavors of lifetime elision. In particular, it's not necessarily obvious that the returned `Cow` has a lifetime attached to `sefl` when the lifetimes are elided.

Reply:
I think this comment applies to all changes in this commit except for `ClassQuery`, which I like.

Reply:
I might be wrong, but I think `fn class(query: ClassQuery)` will trigger a warning once edition lints are enabled. I *think* it is supposed to be spelled `pub fn class(query: ClassQuery<'_>)`, ie, 2018 edition soft-deprecates lifetime elision outside of `&` and `&mut`. Similarly, the best way to spell this fn might be `fn no_expansion(&mut self) -> Option<Cow<'_, [u8]>>`. The `'_` syntax was stablized in 1.26, so I think we can use it.

Reply:
Not a huge fan of the `'_` syntax either to be honest... I like the fact that lifetime parameters are introduced just like type parameters.

I'll likely give in once regex switches to 2018, but I'd prefer to just leave well enough alone for now. :-)